### PR TITLE
Restructured docs for AWS Lambda

### PR DESCRIPTION
**Pull request by AlexanderWert**

### CODE REVIEW

File: docs/monitoring-aws-lambda.asciidoc
Comment (estolfo) [inline docs/monitoring-aws-lambda.asciidoc:20]:
```suggestion
1. To avoid data loss, APM data collected by APM agents needs to be flushed before the execution environment of a lambda function is frozen.
```

File: docs/monitoring-aws-lambda.asciidoc
Comment (estolfo) [inline docs/monitoring-aws-lambda.asciidoc:21]:
```suggestion
2. Flushing APM data must be fast to not impact the response times of lambda function requests.
```

File: docs/monitoring-aws-lambda.asciidoc
Comment (estolfo) [inline docs/monitoring-aws-lambda.asciidoc:25]:
```suggestion
Normally, during the execution of a Lambda function, there's only a single language process running in the AWS Lambda execution environment.  With an AWS Lambda Extension, Lambda users run a _second_ process alongside their main service/application process.
```

File: docs/monitoring-aws-lambda.asciidoc
Comment (estolfo) [inline docs/monitoring-aws-lambda.asciidoc:68]: We also have `ELASTIC_APM_DATA_FORWARDER_TIMEOUT_SECONDS`, `ELASTIC_APM_LOG_LEVEL`, and `ELASTIC_APM_DATA_RECEIVER_SERVER_PORT`

**Review by estolfo — CHANGES_REQUESTED** [review]

File: docs/monitoring-aws-lambda.asciidoc
Comment (AlexanderWert) [reply docs/monitoring-aws-lambda.asciidoc:68]: @estolfo @jlvoiseux  could you add the descriptions for the missing config options here?

Comment (estolfo) [reply docs/monitoring-aws-lambda.asciidoc:68]: I'm trying to remember if we want to actually document `ELASTIC_APM_DATA_RECEIVER_SERVER_PORT` or if we have it for testing. In the meantime, let's not document it.

**Comment by bmorelli25** [comment]
Awesome! A few different things have to happen before this PR can be merged. I'll get started on the following 🚀 

- [ ] 1. Copy the original contents of `aws-lambda-extension.asciidoc` to `monitoring-aws-lambda.asciidoc` and add it to this repo (in `main` and `8.1`).
- [ ] 2. Update this line in the apm-server repo to use the new `monitoring-aws-lambda.asciidoc` file: https://github.com/elastic/apm-server/blame/b951097af2cd622b07b48e1cd77dbd84c1addf54/docs/aws-lambda-extension.asciidoc#L4 (in `main` and `8.1`).
- [ ] 3. Delete the original`aws-lambda-extension.asciidoc` (in `main` and `8.1`).
- [ ] 4. Merge the changes into this PR and fix conflicts
- [ ] 5. Let the doc build run to find broken links.

Then we can come up with a new plan to fix broken links. I'll add additional steps to this comment as the plan evolves.

**Review by bmorelli25 — COMMENTED** [review]
A few comments. Please don't merge yet.

**Review by bmorelli25 — APPROVED** [review]

File: docs/add-extension/code.asciidoc
Comment (bmorelli25) [reply docs/add-extension/code.asciidoc:1]: This is my last comment—I think this PR is ready to merge otherwise. Thanks!

File: docs/monitoring-aws-lambda.asciidoc
Comment (jlvoiseux) [reply docs/monitoring-aws-lambda.asciidoc:75]: Added missing options in [587124e](https://github.com/elastic/apm-aws-lambda/pull/158/commits/587124e607f66d3419e20446382bd67f0f3b8ff1).

File: docs/images/images.asciidoc
Comment (bmorelli25) [inline docs/images/images.asciidoc:3]: FYI @AlexanderWert -- we can't include images across repos, but we can include tagged regions. Adding this file allows us to include the tagged region in the agent docs instead of the image.

**Review by estolfo — APPROVED** [review]

**Review by jlvoiseux — APPROVED** [review]

**Review by bmorelli25 — APPROVED** [review]

**Pull request merged**
